rlp/tests/tests.rs

```diff
@@ -437,4 +437,28 @@ fn test_rlp_is_int() {
 		let rlp = Rlp::new(&data);
 		assert_eq!(rlp.is_int(), false);
 	}
-}
\ No newline at end of file
+}
+
+/// test described in
+///
+/// https://github.com/paritytech/parity-common/issues/48
+#[test]
+fn test_inner_length_capping_for_short_lists() {
+	assert_eq!(Rlp::new(&vec![0xc0 + 0, 0x82, b'a', b'b']).val_at::<String>(0), Err(DecoderError::RlpIsTooShort));
+	assert_eq!(Rlp::new(&vec![0xc0 + 1, 0x82, b'a', b'b']).val_at::<String>(0), Err(DecoderError::RlpIsTooShort));
```

Comment:
it reports `RlpIsTooShort` and not `RlpInconsistentLengthAndData`, because it's trying to take two bytes from a slice that contains only a single byte `&[b'a']`